pkg/core/operation.go

```diff
@@ -245,6 +245,7 @@ func ParseNamespacedOpID(ctx context.Context, nsIDStr string) (string, *fftypes.
 
 type OperationCallbacks interface {
 	OperationUpdate(update *OperationUpdate)
+	BulkOperationUpdates(ctx context.Context, updates []*OperationUpdate, onCommit chan<- error)
```

Comment:
I think passing a channel might be a little sticky. We've gone back and forth on this a few times.

My feeling is a callback function is much easier to pass in, and you then leave the caller the responsibility to decide if they want that callback to use a channel or not.

Reply:
Yes! https://github.com/hyperledger/firefly/pull/1637#issuecomment-2678407214 I got to the same conclusion after reading more code recently and that is what I'm reworking 💪🏼

Reply:
Cool - I think maybe with the discussion below around making the interface synchronous, this might become a moot point